keys/cryptostore/encoder.go

```diff
@@ -16,45 +32,55 @@ var (
 //
 // This should use a well-designed symetric encryption algorithm
 type Encoder interface {
-	Encrypt(key crypto.PrivKey, pass string) ([]byte, error)
-	Decrypt(data []byte, pass string) (crypto.PrivKey, error)
-}
-
-func secret(passphrase string) []byte {
-	// TODO: Sha256(Bcrypt(passphrase))
-	return crypto.Sha256([]byte(passphrase))
+	Encrypt(privKey crypto.PrivKey, passphrase string) (saltBytes []byte, encBytes []byte, err error)
+	Decrypt(saltBytes []byte, encBytes []byte, passphrase string) (privKey crypto.PrivKey, err error)
 }
 
 type secretbox struct{}
 
-func (e secretbox) Encrypt(key crypto.PrivKey, pass string) ([]byte, error) {
-	if pass == "" {
-		return key.Bytes(), nil
+func (e secretbox) Encrypt(privKey crypto.PrivKey, passphrase string) (saltBytes []byte, encBytes []byte, err error) {
+	if passphrase == "" {
```

Comment:
Can't this be deleted, since we don't want users to encrypt keys without a passphrase?

Reply:
We break enough stuff with our encryption library. This is the only way to save keys between versions... export unencrypted and import unencrypted.
We should expose them and use them to migrate keys. Until production release and a stable go-crypto

Reply:
Does this mean its currently possible to migrate keys? We should defntly be trying to support that asap if possible ...